src/token_cache.rs

```diff
@@ -179,3 +180,114 @@ impl TokenCache {
         }
     }
 }
+
+#[cfg(feature = "blocking")]
+#[derive(Clone)]
+pub(crate) struct SyncTokenCache {
+    // (registry, repository, scope) -> (token, expiration)
+    tokens: BTreeMap<TokenCacheKey, TokenCacheValue>,
+    /// Default token expiration in seconds, to use when claim doesn't specify a value
+    pub default_expiration_secs: usize,
+}
+
+#[cfg(feature = "blocking")]
+impl SyncTokenCache {
```

Comment:
Can you take the logic in these functions that is the same and break them out to functions that can be reused rather than copy pasting?